src/http/RequestBuilder.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace markhuot\craftpest\http;
+
+use craft\web\User;
+use markhuot\craftpest\http\requests\GetRequest;
+use markhuot\craftpest\http\requests\PostRequest;
+use markhuot\craftpest\http\requests\WebRequest;
+use markhuot\craftpest\web\Application;
+use markhuot\craftpest\web\TestableResponse;
+use yii\web\Cookie;
+
+class RequestBuilder
+{
+    private WebRequest $request;
+    private Application $app;
+    private RequestHandler $handler;
+
+    public function __construct(
+        string         $method,
+        string         $uri,
+        Application    $app = null,
+        RequestHandler $handler = null,
+    ) {
+        $this->app = $app ?? \Craft::$app;
+        $this->handler = $handler ?? new RequestHandler($this->app);
```

Comment:
We're passing `$app` around a lot but there can really only ever be one. I'd _love_ if Craft allowed us to instantiate multiple apps but today it's not possible (too much static). Do we really need to pass this around today? Or is this future proofing for when Craft better supports this?

Reply:
Mainly to void static calls from inside